docs: update migration guide.

cpojer opened the pull request:
Updated the Migration Guide for Vite 8.

sapphi-red commented on docs/guide/migration.md:
```diff
@@ -230,87 +208,69 @@ export default defineConfig({
 ::::
 
-Note that if you use a plugin that uses `transformWithEsbuild` function, you need to install `esbuild` as a dev dependency as it's now an optional dependency. `transformWithEsbuild` function is now deprecated and will be removed in the future. We recommend to use the new `transformWithOxc` function instead.
+### esbuild Fallbacks
```
```suggestion
#### esbuild Fallbacks
```
I was thinking of this as part of "JavaScript Transforms by Oxc" section. Do you think it should be a separate section?

cpojer commented:
* Updated the "New Features" section so you can copy and paste it into the blog post
* Brought back the link with a new explanation
* Updated heading

sapphi-red commented on docs/guide/migration.md:
```diff
@@ -10,38 +10,36 @@ This section will be moved to the release post before the stable release.
 ### Built-in tsconfig `paths` Support
 
-Vite 8 now has built-in tsconfig `paths` support, thanks to [Oxc Resolver](https://oxc.rs/docs/guide/usage/resolver). This is not enabled by default, because it has a performance cost and is [discouraged by the TypeScript team to use this option to change the behavior of the external tools](https://www.typescriptlang.org/tsconfig/#paths:~:text=Note%20that%20this%20feature%20does%20not%20change%20 ...
+Vite 8 now has built-in support for TypeScript's `paths` option, based on [Oxc Resolver](https://oxc.rs/docs/guide/usage/resolver). This feature has a small performance cost and can [change the behavior of external tools](https://www.typescriptlang.org/tsconfig/#paths:~:text=Note%20that%20this%20feature%20does%20not%20change%20how%20import%20paths%20are%20emitted%20by%20tsc%2C%20so%20paths%20shou ...
```
> can change the behavior of external tools

The sentence itself is valid. But linked docs says a different thing (the external tool should not change the behavior based on this option), so it’s a bit confusing.

sapphi-red reviewed (approved):
Thank you!
